src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -1235,6 +1244,230 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_SetupSrpHost(co
     return error;
 }
 
+#if CHIP_DEVICE_CONFIG_ENABLE_THREAD_DNS_CLIENT
+template <class ImplClass>
+void GenericThreadStackManagerImpl_OpenThread<ImplClass>::FromOtDnsResponseToMdnsData(
+    otDnsServiceInfo & serviceInfo, char * serviceType, chip::Mdns::MdnsService & mdnsService,
+    typename DnsClient::ServiceTxtEntries & serviceTxtEntries)
+{
+    char protocol[chip::Mdns::kMdnsProtocolTextMaxSize + 1];
+
+    // Extract from the <hostname>.default.service.arpa. the <hostname> part.
+    size_t substringSize = strlen(serviceInfo.mHostNameBuffer) - 1 - chip::Mdns::kMdnsDefaultSrpDomainSize - 1;
```

Comment:
Looking at the DiscoveryImplPlatform, it seems that only IP addresses and TXT entries are really used. Maybe you can skip filling the other fields. If not, I think the function should use `strchr` (and the name split by `.`) instead of assuming that the input has a given format.